GeoMesaQuery.md

```diff
@@ -0,0 +1,81 @@
+For objects with extent, their ranges are repeatedly subdivided until ranges with prefixes at least 24 bits in length are found, and the object is associated with each of those prefixes.
+
+## GeoMesa Query Planning ##
+
+### Accumulo ###
+
+In this section we analyze the [following query](https://github.com/geomesa/geomesa-tutorials/blob/293cd73c64b55a23f301065e2e50f696ae6a80bc/geomesa-quickstart-accumulo/src/main/java/com/example/geomesa/accumulo/AccumuloQuickStart.java#L254)
+
+```java
+FeatureIterator featureItr = featureSource.getFeatures(query).features();
+```
+
+assuming that the `SimpleFeature`s that are being looked for are geometries with extent.
+The result of that line is an iterator of `SimpleFeature`s which are responsive to the query.
+
+Restricting attention to query planning, the line above produces the sequence below
+
+1. [AccumuloFeatureSource.getFeatures](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureSource.scala#L87)
+2. [AccumuloFeatureSource.getFeatureSource.getFeaturesNoCache](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureSource.scala#L106-L107)
+3. [AccumuloFeatureCollection constructor](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureSource.scala#L117-L139)
+4. ...
+5. [AccumuloFeatureCollection.reader](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureSource.scala#L172-L173)
+6. [AccumuloDataStore.getFeatureReader](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala#L350-L360)
+7. [AccumuloFeatureReader.apply](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureReader.scala#L47-L58)
+8. [AccumuloFeatureReaderImpl constructor](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureReader.scala#L47-L58)
+9. ...
+10. [QueryPlanner.runQuery](https://github.com/locationtech/geomesa/blob/bab330add6e21ed2c528101d38236a1ca4088c49/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala#L74-L82) 
```

Comment:
Thanks you very much for the feedback.  I am planning to update this PR in the next couple of days.